xdsl/dialects/print.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+from xdsl.parser import Parser
+from xdsl.printer import Printer
+from xdsl.irdl import IRDLOperation, irdl_op_definition, OpAttr, VarOperand
+from xdsl.dialects import builtin
+from xdsl.ir import Dialect, SSAValue, Operation, VerifyException
+
+
+@irdl_op_definition
+class PrintLnOp(IRDLOperation):
+    """
+    A string formatting and printing utility.
+
+    Can be though of as a printf equivalent but with python style format strings.
+
+    Example uses:
+    %42 = arith.constant 42 : i32
+    print.println "The magic number is {}", %42
+    """
+
+    name = "print.println"
+    format_str: OpAttr[builtin.StringAttr]
+
+    format_vals: VarOperand
+
+    def __init__(self, format_str: str, *vals: SSAValue | Operation):
+        super().__init__(
+            operands=[vals], attributes={"format_str": builtin.StringAttr(format_str)}
+        )
+
+    def verify_(self) -> None:
+        num_of_templates = self.format_str.data.count("{}")
+        if not num_of_templates == len(self.format_vals):
+            raise VerifyException(
+                "Number of templates in template string must match number of arguments!"
+            )
+
+    def print(self, printer: Printer):
+        printer.print_string(" ")
+        printer.print_attribute(self.format_str)
+        if len(self.format_vals) > 0:
+            printer.print_string(", ")
+            printer.print_list(self.operands, printer.print_block_argument)
```

Comment:
seems like you'll ned to rewrite your own function here, which sort of seems like a good idea to decouple the IR format of this dialect from the core IR format